lib/rubocop/cop/rails/enum_keyword_args.rb

```diff
@@ -0,0 +1,116 @@
+# frozen_string_literal: true
+
+module RuboCop
+  module Cop
+    module Rails
+      # Looks for enums written with keyword arguments.
+      #
+      # Defining enums with keyword arguments is deprecated
+      # and will be removed in Rails 7.3.
+      #
+      # Positional arguments should be used instead:
+      #
+      # @example
+      #   # bad
+      #   enum status: { active: 0, archived: 1 }, _prefix: true
+      #
+      #   # good
+      #   enum :status, { active: 0, archived: 1 }, prefix: true
+      #
+      class EnumKeywordArgs < Base
```

Comment:
How about the name `Rails/EnumSyntax`?

```suggestion
      class EnumSyntax < Base
```